### PR TITLE
Project board setup scripts added

### LINKED ISSUE #14: Project is not set up automatically
The cloner script fails to set up the project with the error message below.
```bash
Error occurred while setting up project board: 410 {"message": "Projects are disabled for this repository", "documentation_url": "https://docs.github.com/v3/projects"}
Please set up project board manually.
```
I don't know if this is inevitable if we are going to use the board template.

### PR DESCRIPTION
**Pull request by hans-cyfrin**
Changes:
- Added GitHub project board utils.
- Added two new params for the project template ID and the project board title. The template ID can be extracted from the URL of the GitHub project.
- Removed "." in the import statements. It is solely because it gave me errors while working on the feature. Please check carefully.
- Removed the part reading params from config file thinking it's an unnecessary complication given that we already assume .env to be used.
- Revised all params to read default values from .env. The intention is to avoid inputting the source URL, commit hash, and so on repeatedly when the script fails. (maybe not anymore?)

TODO:
- [ ] Revise the README.txt
- [ ] Resolve relative import issue

Closes #14.

### CODE REVIEW

File: audit_repo_cloner/create_audit_repo.py [hunk 1]
```diff
@@ -39,20 +40,24 @@
...
+@click.option("--source-url", help="Source repository URL.", default=os.getenv("SOURCE_REPO_URL"))
+@click.option("--target-repo-name", help="Target repository name (leave blank to use source repo name).", default=os.getenv("TARGET_REPO_NAME"))
+@click.option("--commit-hash", help="Audit commit hash.", default=os.getenv("COMMIT_HASH"))
+@click.option("--auditors", help="Names of the auditors (separated by spaces).", default=os.getenv("ASSIGNED_AUDITORS"))
+@click.option("--github-token", help="Your GitHub developer token to make API calls.", default=os.getenv("GITHUB_ACCESS_TOKEN"))
+@click.option("--organization", help="Your GitHub organization name in which to clone the repo.", default=os.getenv("GITHUB_ORGANIZATION"))
+@click.option("--project-template-id", help="ID of the GitHub project board template.", default=os.getenv("PROJECT_TEMPLATE_ID"))
+@click.option("--project-title", help="Title of the new project board on GitHub.", default=os.getenv("PROJECT_TITLE"))
 def create_audit_repo(
     config: str = "",
```
Comment (giovannidisiena) [inline audit_repo_cloner/create_audit_repo.py:52]:
```suggestion
```

File: audit_repo_cloner/create_audit_repo.py [hunk 2]
```diff
@@ -39,20 +40,24 @@
     prog_name=__title__,
 )
 @click.option("--config", type=click.Path(exists=True), help="Path to YAML config file")
```
Comment (giovannidisiena) [inline audit_repo_cloner/create_audit_repo.py:42]:
```suggestion
```

File: audit_repo_cloner/github_project_utils.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    response = requests.request("GET", url, headers=headers, data=payload)
+    res = response.json()
+    return res.get('node_id', '')
+
+def get_project_node_id(token: str, org_name:str, project_template_id: str)->str:
+    """
+    Get the GitHub organization's node id to use for later call to GraphQL
+        token (str): GitHub personal access token
+        org_name (str): GitHub organization name (username)
+        project_template_id (str): ID of the project template, can be extracted from the link (e.g. https://github.com/orgs/KupiaSec/projects/7/views/2 => 7 is the ID)
```
Comment (giovannidisiena) [inline audit_repo_cloner/github_project_utils.py:30]:
```suggestion
        project_template_id (str): ID of the project template, can be extracted from the link (e.g. https://github.com/orgs/Cyfrin/projects/7/views/2 => 7 is the ID)
```

File: audit_repo_cloner/github_project_utils.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    res = response.json()
+    return res.get('data', {}).get('organization', {}).get('projectV2', {}).get('id', '')
+
+
+def clone_project(token: str, org_name: str, project_template_id:str, project_title:str = '')->str:
+    """
+    Clone a GitHub project from the template
+        token (str): GitHub personal access token
+        org_node_id (str): GitHub Organization node ID (can be retrieved by calling get_organization_node_id)
+        project_template_id (str): ID of the project template, can be extracted from the link (e.g. https://github.com/orgs/KupiaSec/projects/7/views/2 => 7 is the ID)
```
Comment (giovannidisiena) [inline audit_repo_cloner/github_project_utils.py:50]:
```suggestion
        project_template_id (str): ID of the project template, can be extracted from the link (e.g. https://github.com/orgs/Cyfrin/projects/7/views/2 => 7 is the ID)
```

**Review by giovannidisiena — COMMENTED** [review]
I didn't think to copy the template project - awesome!

Just a couple of minor nitpick suggestions.

Also, I'm wondering whether it is cleaner to use a GraphQL client like [here](https://github.com/Cyfrin/audit-repo-cloner/commit/904a9c8f811b1a373ed115468fff13d23be33924) rather than HTTP requests. In my opinion, it's easier to read and make sense of the mutations. What do you think? The logic is practically the same, except the retrieval of node ids is not currently separated into functions and the copy mutation would need to be added in place of the `TODO` comment.

@bensacc are you able to look into the relative import issue, please? It seems we can currently only satisfy either the pip package or running the cli command from source. Perhaps this is due to how the package is currently set up, but there shouldn’t be any reason we have to choose one or the other right?

**Comment by hans-cyfrin** [comment]
@giovannidisiena It would be absolutely better to use GraphQL instead of HTTP request here.
I am not familiar with GraphQL and that's why I didn't ever think of using it directly.

**Comment by giovannidisiena** [comment]
@hans-cyfrin great, that’s not a problem - I will add it to my todo list!

**Comment by giovannidisiena** [comment]
@hans-cyfrin everything should be working now. If you agree then all that's left is to update the README. We can re-visit the relative import issue another time if/when we decide to fix the package.

**Pull request merged**